taskt/Core/Automation/Commands/UIAutomationCommand.cs

```diff
@@ -206,125 +207,248 @@ public AutomationElement SearchForGUIElement(object sender, string variableWindo
         }
         public override void RunCommand(object sender)
         {
-
             var engine = (Core.Automation.Engine.AutomationEngineInstance)sender;
-
             //create variable window name
             var variableWindowName = v_WindowName.ConvertToUserVariable(sender);
-
             if (variableWindowName == "Current Window")
-            {
                 variableWindowName = User32Functions.GetActiveWindowTitle();
-            }
 
-            var requiredHandle =  SearchForGUIElement(sender, variableWindowName);
+            dynamic requiredHandle = null;
+            if (v_AutomationType != "Wait For Element To Exist")
+                requiredHandle =  SearchForGUIElement(sender, variableWindowName);
 
-
-            //if element exists type
-            if (v_AutomationType == "Check If Element Exists")
+            switch (v_AutomationType)
```

Comment:
It may be worth splitting these cases out into separate methods for clarity. If the necessary parameters would be unwieldy we can keep the current approach.

Reply:
I kind of agree with you, but for now I was following the same conventions used in the web browser automation command.

Reply:
Gotcha. We can defer it to a later pass of the general command library. Thank you!